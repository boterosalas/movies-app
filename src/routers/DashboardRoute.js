import React, { useContext, useEffect } from 'react';
import { Redirect, Route, Switch, useHistory } from 'react-router-dom';
import Footer from '../components/shared/footer/Footer';
import HomeScreen from '../components/home/HomeScreen';
import MovieScreen from '../components/movie/MovieScreen';
import SearchScreen from '../components/search/SearchScreen';
import Navbar from '../components/shared/navbar/Navbar';
import { UserContext } from './UserContext';
import './DashboardRoute.css';

const DashboardRoute = () => {
    let history = useHistory();
    const { user } = useContext(UserContext);
    useEffect(() => {
        if (!user) {
            history.push('/login');
        }
    }, [user, history])
    return (
        <>
            {
                user &&
                <>
                    <Navbar />
                    <main className="container mt-7">
                        <Switch>
                            <Route exact path="/" component={HomeScreen} />
                            <Route exact path="/search/:text" component={SearchScreen} />
                            <Route exact path="/movie/:id" component={MovieScreen} />
                            <Redirect to="/" />
                        </Switch>
                    </main>
                    <Footer />
                </>
            }
        </>
    )
}

export default DashboardRoute
