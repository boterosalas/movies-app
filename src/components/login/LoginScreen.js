import React, { useContext, useState } from 'react';
import { UserContext } from '../../routers/UserContext';
import './Login.css';

const LoginScreen = ({ history }) => {
    const { setUser } = useContext(UserContext);
    const [formState, setFormState] = useState({
        name: '',
        email: ''
    });

    const { name, email } = formState;

    const handleInputChange = ({ target }) => {
        setFormState({
            ...formState,
            [target.name]: target.value
        })
    }

    const handleLogin = (e) => {
        e.preventDefault();
        setUser(formState);
        history.replace('/');
    }

    return (
        <div className="principal-container">
            <h1 className="text-dark">Inicio de sesi√≥n</h1>
            <form className="form" onSubmit={handleLogin}>
                <div className="mb-3">
                    <label htmlFor="inputName" className="form-label">Nombre</label>
                    <input onChange={handleInputChange} value={name} name="name" placeholder="John Doe" type="text" className="form-control" id="inputName" />
                </div>
                <div className="mb-3">
                    <label htmlFor="inputEmail" className="form-label">Correo</label>
                    <input onChange={handleInputChange} value={email} name="email" placeholder="johndoe@example.com" type="email" className="form-control" id="inputEmail" />
                </div>
                <button type="submit" className="btn btn-primary btn-block">Ingresar</button>
            </form>
        </div>
    )
}

export default LoginScreen
