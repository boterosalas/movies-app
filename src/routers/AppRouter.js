import React, { useState } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route
} from "react-router-dom";
import LoginScreen from '../components/login/LoginScreen';
import DashboardRoute from './DashboardRoute';
import { UserContext } from './UserContext';


const AppRouter = () => {
    const [user, setUser] = useState(null);
    return (
        <UserContext.Provider value={{ user, setUser }} >
            <Router>
                <div>
                    <Switch>
                        <Route exac path="/login" component={LoginScreen} />
                        <Route exac path="/" component={DashboardRoute} />
                    </Switch>
                </div>
            </Router>
        </UserContext.Provider>
    )
}

export default AppRouter
