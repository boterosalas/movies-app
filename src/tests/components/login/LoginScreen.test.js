import React from 'react';
import { mount } from 'enzyme';
import '@testing-library/jest-dom';
import LoginScreen from '../../../components/login/LoginScreen';
import { MemoryRouter, Router } from 'react-router-dom';
import { UserContext } from '../../../routers/UserContext';


describe('Test on <LoginScreen />', () => {

    const setUser = jest.fn();

    const user = {
        name: 'David',
        email: 'boterosalas@gmail.com'
    };

    const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        replace: jest.fn()
    };

    const wrapper = mount(
        <UserContext.Provider value={{ setUser }} >
            <MemoryRouter>
                <Router history={historyMock}>
                    <LoginScreen />
                </Router>
            </MemoryRouter>
        </UserContext.Provider>
    );


    test('Should display the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should redirect to <HomeScreen /> and set User', () => {
        wrapper.find('#inputName').simulate('change', { target: { name: 'name', value: user.name } });
        wrapper.find('#inputEmail').simulate('change', { target: { name: 'email', value: user.email } });
        wrapper.find('form').simulate('submit');
        expect(setUser).toHaveBeenCalledWith(user);
        expect(historyMock.replace).toHaveBeenCalledWith('/')
    });

})