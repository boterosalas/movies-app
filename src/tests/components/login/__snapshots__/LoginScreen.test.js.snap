// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <LoginScreen /> Should display the component correctly 1`] = `
<div
  className="principal-container"
>
  <h1
    className="text-dark"
  >
    Inicio de sesi√≥n
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="inputName"
      >
        Nombre
      </label>
      <input
        className="form-control"
        id="inputName"
        name="name"
        onChange={[Function]}
        placeholder="John Doe"
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="inputEmail"
      >
        Correo
      </label>
      <input
        className="form-control"
        id="inputEmail"
        name="email"
        onChange={[Function]}
        placeholder="johndoe@example.com"
        type="email"
        value=""
      />
    </div>
    <button
      className="btn btn-primary btn-block"
      type="submit"
    >
      Ingresar
    </button>
  </form>
</div>
`;
